### PR TITLE
[#384] Add check of Notificator interface value in Initialize func

### LINKED ISSUE #384: S3 GW panics when notifications are not enabled
The app panics in `layer/layer.go:294` when calls `c.Listen(ctx)` because Notificator interface value `c` is `nil`

### PR DESCRIPTION
**Pull request by masterSplinter01**
Closes #384

### CODE REVIEW

File: api/layer/layer.go [hunk 1]
```diff
@@ -281,7 +282,7 @@ func (n *layer) EphemeralKey() *keys.PublicKey {
 }
 
 func (n *layer) Initialize(ctx context.Context, c Notificator) error {
-	if c == nil {
+	if c == nil || reflect.ValueOf(c).Kind() != reflect.Struct && reflect.ValueOf(c).IsNil() {
```
Comment (alexvanin) [inline api/layer/layer.go:285]: Can we avoid reflection until it is absolutely necessary?
We can cast interface to exact structure and check nil value there, can't we?

**Comment by KirillovDenis** [comment]
Actually, I don't like use `reflect`. Can we do refactor to avoid this.
For example we can use something like (during call `Initialize` method):
```
	if v.GetBool(cfgEnableNATS) {
		nopts := getNotificationsOptions(v, l)
		nc, err := notifications.NewController(nopts, l)
		if err != nil {
			l.Fatal("failed to enable notifications", zap.Error(err))
		}
		if err = obj.Initialize(ctx, nc); err != nil {
			l.Fatal("couldn't initialize layer", zap.Error(err))
		}
	}
```

File: api/layer/layer.go (hunk 1 above)
Comment (KirillovDenis) [reply api/layer/layer.go:285]: Then we need know exactly structure type/s. It seems that using an interface as a parameter makes no sense in this case

Comment (masterSplinter01) [reply api/layer/layer.go]: Why we should avoid using of reflect?

**Comment by KirillovDenis** [comment]
We can now drop the following condition in `getNotificationsOptions` function :
```
if enabled := v.GetBool(cfgEnableNATS); !enabled {
	return nil
}
```

Also we can drop the following condition in `NewController` function:
```
if p == nil {
	return nil, nil
}
```

File: api/layer/layer.go (hunk 1 above)
Comment (KirillovDenis) [reply api/layer/layer.go]: It's like a marker that our code does something unsafe and dirty though we actually don't do this. It also makes the code less readable.

**Review by KirillovDenis — APPROVED** [review]

**Pull request merged**